sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -663,11 +663,13 @@ case class UnresolvedWith(
  *                                   predicates that have been pushed down into `child`. This is
  *                                   a temporary field used by optimization rules for CTE predicate
  *                                   pushdown to help ensure rule idempotency.
+ * @param mergedScalarSubquery If this definition is a merged scalar subquery.
```

Comment:
I think it's better to clarify what this flag means to CTE relation, instead of linking to another feature. how about
```
@param underSubquery If true, it means we don't need to add a shuffle for this CTE relation as subquery reuse will kick in.
```

Reply:
Sounds good, fixed in https://github.com/apache/spark/pull/32298/commits/19128ff010072979da7d416422330bf96699e24b.